src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  return items.reduce((grouped: GroupsMap<T>, item: T) => {
+    const value = String(item[key]);
+
+    return {
+      ...grouped, [value]: [...(grouped[value] || []), item],
+    };
```

Comment:
```suggestion
    return {
      ...grouped, 
      [value]: [...(grouped[value] || []), item],
    };
```
It would be also great to create a variable for an array to make your code a bit cleaner